virtwho/configure.py

```diff
@@ -1,4 +1,134 @@
+            self.update('type', 'libvirt')
+        else:
+            self.update('type', mode)
+            self.update('hypervisor_id', 'hostname')
+            self.update('owner', self.register.default_org)
+        if mode is 'kubevirt':
+            self.update('kubeconfig', self.hypervisor.config_file)
+        if mode in ('esx', 'xen', 'hyperv', 'rhevm', 'libvirt'):
+            self.update('server', self.hypervisor.server)
+            self.update('username', self.hypervisor.username)
+            self.update('password', self.hypervisor.password)
+
+    def update(self, option, value):
+        """Add or update an option
+        :param option: Option will be added if not exist.
+        :param value: Value to update for the option.
+        """
+        self.cfg.update(self.section, option, value)
+
+    def delete(self, option):
+        """Delete an option
+        :param option: Option to remove.
+        """
+        self.cfg.delete(self.section, option)
+
+    def destroy(self):
+        """Remove both the local and remote files"""
+        os.remove(self.local_file)
+        self.remote_ssh.remove_file(self.remote_file)
+
+
+class VirtwhoGlobalConfig:
+    """Able to manage /etc/virt-who.conf file when call this class"""
+
+    def __init__(self, mode=None):
+        """virt-who.conf file is backed up to /temp directory of the
+        project root.
+        :param mode: Hypervisor mode. When mode is local, will manage
+            the local libvirt host, othervise will manage the host for
+            all other remote modes.
+        """
+        self.mode = mode
+        self.remote_ssh = virtwho_ssh_connect(self.mode)
+        self.local_file = os.path.join(TEMP_DIR, 'virt-who.conf')
+        self.remote_file = '/etc/virt-who.conf'
+        self.save_file = os.path.join(TEMP_DIR, 'virt-who.conf.save')
+        if not os.path.exists(self.save_file):
+            self.remote_ssh.get_file(self.remote_file, self.save_file)
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+
+    def update(self, section, option, value):
+        """Add section, add option or update option
+        :param section: Section will be added if not exist.
+        :param option: Option will be added if not exist.
+        :param value: Value to update for the option.
+        """
+        self.cfg.update(section, option, value)
+
+    def delete(self, section, option=None):
+        """Remove a section or option
+        :param section: Section will be removed when no option provided.
+        :param option: Option to remove.
+        """
+        self.cfg.delete(section, option)
+
+
+def virtwho_ssh_connect(mode=None):
+    """Define the ssh connection of virt-who host, get data from
+    virtwho.ini file.
+    :param mode: The test hypervisor mode.
+    """
+    virtwho = config.virtwho
+    if mode == 'local':
+        virtwho = config.local
+        port = 22
+    else:
+        port = virtwho.port or 22
```

Comment:
one line is enough:
```port = 22 or virtwho.port```

port=22 must be first, it means define port=22 first, if the value(such as 33242) of virtwho.port is provided, port will be re-define to 33242.

Reply:
ok, I have tried this, it's really, so that I think we can define only one port is ok:
```
    virtwho = config.virtwho
    if mode == 'local':
        virtwho = config.local
    port = 22 or virtwho.port
    host = virtwho.server
    username = virtwho.username
    password = virtwho.password
```

Two things need to update:
1). update virtwho.ini, use `server` to replace the `host` in virtwho section
2). add a new option `port` for `local` section`

Reply:
yeah, also found the gap for ```server``` and ```host```, updated it in another pr.
and also added ```port``` for ```local``` section.